platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -911,7 +912,9 @@ public void visitDependencies(TaskDependencyResolveContext context) {
     @Override
     public TaskDependency getTaskDependencyFromProjectDependency(final boolean useDependedOn, final String taskName) {
         if (useDependedOn) {
-            return new TasksFromProjectDependencies(taskName, this::getAllDependencies, taskDependencyFactory);
+            return new TasksFromProjectDependencies(taskName, () -> {
```

Comment:
❓ Why change this constructor from accepting a `DependencySet` and doing the filtering to only `ProjectDependency`s internally?

Reply:
It made testing a little easier in `TasksFromProjectDependenciesTest` to avoid needing mocks